File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        unordered_map<int, int> m;

        for (int i = 0; i < numbers.size(); i ++) {
            int complement = target - numbers[i];
            if (m.count(complement)) {
                return {m[complement] + 1, i + 1};
            }

            m[numbers[i]] = i;
        }

        return {};
    }
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        a, b = 0, len(numbers) - 1

        while a < b:
            current_sum = numbers[a] + numbers[b]
            if current_sum == target: return [a + 1, b + 1]
            elif current_sum < target: a += 1
            else: b -= 1

        return [-1, -1] 
        